README.md
# Gilbert car shop

## Installation

requires [Node.js](https://nodejs.org/) v10+ and jdk1.8 and local [mongodb](https://www.mongodb.com/try/download/community) community server to run.

Install the dependencies and devDependencies and start the server.

```sh
cd gilber-car-shop
cd frontend
npm i
npm run build
cd ..
gradlew bootrun
```

Add following data in mongodb server under a database name 'testdata' or update the application.properties
```sh
 [
        {
          "_id": "1",
          "name": "Warehouse A",
          "location": {
            "lat": "47.13111",
            "long": "-61.54801"
          },
          "cars": {
            "location": "West wing",
            "vehicles": [
              {
                "_id": 1,
                "make": "Volkswagen",
                "model": "Jetta III",
                "year_model": 1995,
                "price": 12947.52,
                "licensed": true,
                "date_added": "2018-09-18"
              },
              {
                "_id": 2,
                "make": "Chevrolet",
                "model": "Corvette",
                "year_model": 2004,
                "price": 20019.64,
                "licensed": true,
                "date_added": "2018-01-27"
              },
              {
                "_id": 3,
                "make": "Ford",
                "model": "Expedition EL",
                "year_model": 2008,
                "price": 27323.42,
                "licensed": false,
                "date_added": "2018-07-03"
              },
              {
                "_id": 4,
                "make": "Infiniti",
                "model": "FX",
                "year_model": 2010,
                "price": 8541.62,
                "licensed": true,
                "date_added": "2018-03-23"
              },
              {
                "_id": 5,
                "make": "GMC",
                "model": "Safari",
                "year_model": 1998,
                "price": 14772.5,
                "licensed": false,
                "date_added": "2018-07-04"
              },
              {
                "_id": 6,
                "make": "Plymouth",
                "model": "Colt Vista",
                "year_model": 1994,
                "price": 6642.45,
                "licensed": true,
                "date_added": "2018-07-11"
              },
              {
                "_id": 7,
                "make": "Cadillac",
                "model": "Escalade ESV",
                "year_model": 2008,
                "price": 24925.75,
                "licensed": false,
                "date_added": "2018-01-29"
              },
              {
                "_id": 8,
                "make": "Mitsubishi",
                "model": "Pajero",
                "year_model": 2002,
                "price": 28619.45,
                "licensed": false,
                "date_added": "2018-06-12"
              },
              {
                "_id": 9,
                "make": "Infiniti",
                "model": "Q",
                "year_model": 1995,
                "price": 6103.4,
                "licensed": false,
                "date_added": "2017-11-13"
              },
              {
                "_id": 10,
                "make": "Ford",
                "model": "Mustang",
                "year_model": 1993,
                "price": 18992.7,
                "licensed": false,
                "date_added": "2018-01-29"
              },
              {
                "_id": 11,
                "make": "Chevrolet",
                "model": "G-Series 1500",
                "year_model": 1997,
                "price": 23362.41,
                "licensed": false,
                "date_added": "2018-07-30"
              },
              {
                "_id": 12,
                "make": "Cadillac",
                "model": "DeVille",
                "year_model": 1993,
                "price": 18371.53,
                "licensed": false,
                "date_added": "2018-01-24"
              },
              {
                "_id": 13,
                "make": "Acura",
                "model": "NSX",
                "year_model": 2001,
                "price": 23175.76,
                "licensed": false,
                "date_added": "2018-03-28"
              },
              {
                "_id": 14,
                "make": "Ford",
                "model": "Econoline E250",
                "year_model": 1994,
                "price": 26605.54,
                "licensed": true,
                "date_added": "2018-05-13"
              },
              {
                "_id": 15,
                "make": "Lexus",
                "model": "GX",
                "year_model": 2005,
                "price": 27395.26,
                "licensed": false,
                "date_added": "2017-11-12"
              },
              {
                "_id": 16,
                "make": "Dodge",
                "model": "Ram Van 3500",
                "year_model": 1999,
                "price": 6244.51,
                "licensed": true,
                "date_added": "2018-09-28"
              },
              {
                "_id": 17,
                "make": "Dodge",
                "model": "Caravan",
                "year_model": 1995,
                "price": 16145.27,
                "licensed": false,
                "date_added": "2017-11-25"
              },
              {
                "_id": 18,
                "make": "Dodge",
                "model": "Dynasty",
                "year_model": 1992,
                "price": 15103.84,
                "licensed": true,
                "date_added": "2018-08-12"
              },
              {
                "_id": 19,
                "make": "Dodge",
                "model": "Ram 1500",
                "year_model": 2004,
                "price": 9977.65,
                "licensed": true,
                "date_added": "2018-01-18"
              }
            ]
          }
        },
        {
          "_id": "2",
          "name": "Warehouse B",
          "location": {
            "lat": "15.95386",
            "long": "7.06246"
          },
          "cars": {
            "location": "East wing",
            "vehicles": [
              {
                "_id": 20,
                "make": "Maserati",
                "model": "Coupe",
                "year_model": 2005,
                "price": 19957.71,
                "licensed": false,
                "date_added": "2017-11-14"
              },
              {
                "_id": 21,
                "make": "Isuzu",
                "model": "Rodeo",
                "year_model": 1998,
                "price": 6303.99,
                "licensed": false,
                "date_added": "2017-12-03"
              },
              {
                "_id": 22,
                "make": "Infiniti",
                "model": "I",
                "year_model": 2002,
                "price": 6910.16,
                "licensed": false,
                "date_added": "2017-10-15"
              },
              {
                "_id": 23,
                "make": "Nissan",
                "model": "Altima",
                "year_model": 1994,
                "price": 8252.66,
                "licensed": false,
                "date_added": "2018-08-12"
              },
              {
                "_id": 24,
                "make": "Toyota",
                "model": "Corolla",
                "year_model": 2009,
                "price": 23732.11,
                "licensed": false,
                "date_added": "2018-02-13"
              },
              {
                "_id": 25,
                "make": "Acura",
                "model": "MDX",
                "year_model": 2011,
                "price": 8487.19,
                "licensed": false,
                "date_added": "2018-04-18"
              },
              {
                "_id": 26,
                "make": "BMW",
                "model": "7 Series",
                "year_model": 1998,
                "price": 29069.52,
                "licensed": false,
                "date_added": "2017-10-29"
              },
              {
                "_id": 27,
                "make": "Nissan",
                "model": "Maxima",
                "year_model": 2004,
                "price": 11187.68,
                "licensed": false,
                "date_added": "2018-07-16"
              },
              {
                "_id": 28,
                "make": "Audi",
                "model": "A8",
                "year_model": 1999,
                "price": 16047.9,
                "licensed": false,
                "date_added": "2017-12-05"
              },
              {
                "_id": 29,
                "make": "Nissan",
                "model": "Murano",
                "year_model": 2005,
                "price": 25859.78,
                "licensed": false,
                "date_added": "2018-06-06"
              },
              {
                "_id": 30,
                "make": "Acura",
                "model": "RL",
                "year_model": 2010,
                "price": 13232.13,
                "licensed": true,
                "date_added": "2017-12-16"
              },
              {
                "_id": 31,
                "make": "Mitsubishi",
                "model": "Chariot",
                "year_model": 1987,
                "price": 15665.23,
                "licensed": false,
                "date_added": "2018-02-21"
              },
              {
                "_id": 32,
                "make": "GMC",
                "model": "3500 Club Coupe",
                "year_model": 1992,
                "price": 18129.37,
                "licensed": true,
                "date_added": "2018-09-23"
              },
              {
                "_id": 33,
                "make": "Dodge",
                "model": "Dakota",
                "year_model": 2009,
                "price": 14479.37,
                "licensed": false,
                "date_added": "2017-12-12"
              },
              {
                "_id": 34,
                "make": "Mercury",
                "model": "Grand Marquis",
                "year_model": 1996,
                "price": 20614.72,
                "licensed": false,
                "date_added": "2018-05-26"
              },
              {
                "_id": 35,
                "make": "Kia",
                "model": "Sportage",
                "year_model": 2001,
                "price": 27106.47,
                "licensed": false,
                "date_added": "2018-03-14"
              },
              {
                "_id": 36,
                "make": "Chevrolet",
                "model": "Blazer",
                "year_model": 1994,
                "price": 14835.48,
                "licensed": false,
                "date_added": "2017-11-10"
              },
              {
                "_id": 37,
                "make": "Mercedes-Benz",
                "model": "SL-Class",
                "year_model": 1994,
                "price": 27717.28,
                "licensed": false,
                "date_added": "2018-08-17"
              },
              {
                "_id": 38,
                "make": "Honda",
                "model": "Civic Si",
                "year_model": 2003,
                "price": 18569.86,
                "licensed": true,
                "date_added": "2018-09-12"
              },
              {
                "_id": 39,
                "make": "Mercedes-Benz",
                "model": "CL-Class",
                "year_model": 2002,
                "price": 23494.78,
                "licensed": true,
                "date_added": "2018-05-24"
              },
              {
                "_id": 40,
                "make": "Volkswagen",
                "model": "Jetta",
                "year_model": 2006,
                "price": 25469.49,
                "licensed": false,
                "date_added": "2018-04-23"
              },
              {
                "_id": 41,
                "make": "Pontiac",
                "model": "Grand Prix",
                "year_model": 1975,
                "price": 11600.74,
                "licensed": true,
                "date_added": "2018-01-14"
              },
              {
                "_id": 42,
                "make": "Infiniti",
                "model": "FX",
                "year_model": 2012,
                "price": 22000.62,
                "licensed": true,
                "date_added": "2018-06-09"
              },
              {
                "_id": 43,
                "make": "Jaguar",
                "model": "XK",
                "year_model": 2006,
                "price": 10260.79,
                "licensed": true,
                "date_added": "2018-09-28"
              },
              {
                "_id": 44,
                "make": "Cadillac",
                "model": "STS",
                "year_model": 2007,
                "price": 13740.2,
                "licensed": false,
                "date_added": "2018-02-25"
              },
              {
                "_id": 45,
                "make": "Pontiac",
                "model": "Sunfire",
                "year_model": 1997,
                "price": 28489.1,
                "licensed": false,
                "date_added": "2017-12-05"
              },
              {
                "_id": 46,
                "make": "Cadillac",
                "model": "SRX",
                "year_model": 2004,
                "price": 26750.38,
                "licensed": true,
                "date_added": "2018-08-07"
              },
              {
                "_id": 47,
                "make": "Land Rover",
                "model": "Freelander",
                "year_model": 2004,
                "price": 21770.59,
                "licensed": false,
                "date_added": "2018-09-01"
              },
              {
                "_id": 48,
                "make": "Suzuki",
                "model": "Forenza",
                "year_model": 2005,
                "price": 28834.26,
                "licensed": false,
                "date_added": "2018-05-09"
              },
              {
                "_id": 49,
                "make": "Saab",
                "model": "9-7X",
                "year_model": 2005,
                "price": 25975.68,
                "licensed": false,
                "date_added": "2018-06-07"
              },
              {
                "_id": 50,
                "make": "Ford",
                "model": "Fusion",
                "year_model": 2012,
                "price": 28091.96,
                "licensed": false,
                "date_added": "2018-07-15"
              }
            ]
          }
        },
        {
          "_id": "3",
          "name": "Warehouse C",
          "location": {
            "lat": "39.12788",
            "long": "-2.71398"
          },
          "cars": {
            "location": "Suid wing",
            "vehicles": [
              {
                "_id": 51,
                "make": "Cadillac",
                "model": "Escalade",
                "year_model": 2005,
                "price": 7429.18,
                "licensed": true,
                "date_added": "2018-09-26"
              },
              {
                "_id": 52,
                "make": "Porsche",
                "model": "Cayenne",
                "year_model": 2011,
                "price": 17066.31,
                "licensed": true,
                "date_added": "2017-10-19"
              },
              {
                "_id": 53,
                "make": "Mercedes-Benz",
                "model": "SL-Class",
                "year_model": 2005,
                "price": 14066.06,
                "licensed": false,
                "date_added": "2018-08-08"
              },
              {
                "_id": 54,
                "make": "Mitsubishi",
                "model": "RVR",
                "year_model": 1995,
                "price": 22560.18,
                "licensed": false,
                "date_added": "2018-05-25"
              },
              {
                "_id": 55,
                "make": "Volvo",
                "model": "850",
                "year_model": 1995,
                "price": 25762.08,
                "licensed": true,
                "date_added": "2017-10-03"
              },
              {
                "_id": 56,
                "make": "Honda",
                "model": "del Sol",
                "year_model": 1994,
                "price": 18840.96,
                "licensed": true,
                "date_added": "2017-11-25"
              },
              {
                "_id": 57,
                "make": "Infiniti",
                "model": "Q",
                "year_model": 1996,
                "price": 28773.14,
                "licensed": true,
                "date_added": "2018-08-09"
              },
              {
                "_id": 58,
                "make": "Mercedes-Benz",
                "model": "500E",
                "year_model": 1992,
                "price": 17141.08,
                "licensed": true,
                "date_added": "2018-08-13"
              },
              {
                "_id": 59,
                "make": "GMC",
                "model": "Envoy XL",
                "year_model": 2002,
                "price": 18983.52,
                "licensed": true,
                "date_added": "2018-03-14"
              },
              {
                "_id": 60,
                "make": "Volkswagen",
                "model": "Touareg 2",
                "year_model": 2008,
                "price": 15611.22,
                "licensed": true,
                "date_added": "2018-02-22"
              }
            ]
          }
        },
        {
          "_id": "4",
          "name": "Warehouse D",
          "location": {
            "lat": "-70.84354",
            "long": "132.22345"
          },
          "cars": {
            "location": "Suid wing",
            "vehicles": [
              {
                "_id": 61,
                "make": "Saab",
                "model": "900",
                "year_model": 1987,
                "price": 8771.0,
                "licensed": false,
                "date_added": "2017-12-01"
              },
              {
                "_id": 62,
                "make": "Mazda",
                "model": "B-Series",
                "year_model": 1998,
                "price": 23407.59,
                "licensed": false,
                "date_added": "2018-03-01"
              },
              {
                "_id": 63,
                "make": "GMC",
                "model": "Sierra 3500",
                "year_model": 2012,
                "price": 5869.23,
                "licensed": true,
                "date_added": "2018-04-27"
              },
              {
                "_id": 64,
                "make": "Chevrolet",
                "model": "Corvette",
                "year_model": 1964,
                "price": 16630.67,
                "licensed": true,
                "date_added": "2018-05-31"
              },
              {
                "_id": 65,
                "make": "Toyota",
                "model": "Tacoma",
                "year_model": 1997,
                "price": 11597.18,
                "licensed": false,
                "date_added": "2018-03-30"
              },
              {
                "_id": 66,
                "make": "GMC",
                "model": "Sonoma",
                "year_model": 2004,
                "price": 18248.21,
                "licensed": false,
                "date_added": "2018-03-09"
              },
              {
                "_id": 67,
                "make": "Bugatti",
                "model": "Veyron",
                "year_model": 2009,
                "price": 8051.64,
                "licensed": false,
                "date_added": "2018-01-10"
              },
              {
                "_id": 68,
                "make": "Dodge",
                "model": "Ram 1500 Club",
                "year_model": 1996,
                "price": 14008.3,
                "licensed": false,
                "date_added": "2018-05-01"
              },
              {
                "_id": 69,
                "make": "Land Rover",
                "model": "Discovery Series II",
                "year_model": 2001,
                "price": 18620.19,
                "licensed": false,
                "date_added": "2018-03-03"
              },
              {
                "_id": 70,
                "make": "Volvo",
                "model": "960",
                "year_model": 1993,
                "price": 7316.93,
                "licensed": true,
                "date_added": "2018-02-15"
              },
              {
                "_id": 71,
                "make": "Chrysler",
                "model": "LHS",
                "year_model": 2001,
                "price": 29444.71,
                "licensed": false,
                "date_added": "2017-10-25"
              },
              {
                "_id": 72,
                "make": "Porsche",
                "model": "944",
                "year_model": 1984,
                "price": 7396.95,
                "licensed": true,
                "date_added": "2017-10-26"
              },
              {
                "_id": 73,
                "make": "Subaru",
                "model": "Legacy",
                "year_model": 2010,
                "price": 24491.8,
                "licensed": false,
                "date_added": "2017-12-26"
              },
              {
                "_id": 74,
                "make": "Volvo",
                "model": "XC90",
                "year_model": 2003,
                "price": 29009.65,
                "licensed": true,
                "date_added": "2018-04-24"
              },
              {
                "_id": 75,
                "make": "Buick",
                "model": "Skyhawk",
                "year_model": 1985,
                "price": 10567.27,
                "licensed": false,
                "date_added": "2018-03-21"
              },
              {
                "_id": 76,
                "make": "GMC",
                "model": "Envoy XUV",
                "year_model": 2004,
                "price": 20997.71,
                "licensed": true,
                "date_added": "2018-03-27"
              },
              {
                "_id": 77,
                "make": "Volvo",
                "model": "S60",
                "year_model": 2009,
                "price": 28614.95,
                "licensed": false,
                "date_added": "2018-07-25"
              },
              {
                "_id": 78,
                "make": "Pontiac",
                "model": "Montana",
                "year_model": 2000,
                "price": 11221.14,
                "licensed": false,
                "date_added": "2018-01-04"
              },
              {
                "_id": 79,
                "make": "Lexus",
                "model": "RX",
                "year_model": 2002,
                "price": 23194.01,
                "licensed": false,
                "date_added": "2018-05-02"
              },
              {
                "_id": 80,
                "make": "Lexus",
                "model": "RX",
                "year_model": 2000,
                "price": 17805.45,
                "licensed": false,
                "date_added": "2018-09-11"
              }
            ]
          }
        }
      ]
```


CHANGELOG.md
# Changelog

**v1.0**
- feat: List all cars across all warehouses

**v2.0**
- feat: Show details of car on click of 'Show More
- feat: Accessiblity improvement for the website

**v3.0**
- feat: Able to put car in shopping basket and show total value

**v4.0**
- feat: refactored code to make project scalable

**v5.0**
- feat: add redux thunk middle ware for ajax calls in store

frontend/mock-data/cars.js
export const cars = [
  {
    "id": 1,
    "location": {
      "name": "West wing",
      "warehouse": "Warehouse A",
      "lat": "47.13111",
      "long": "-61.54801"
    },
    "carInfo": {
      "make": "Volkswagen",
      "model": "Jetta III",
      "yearModel": 1995,
      "price": 12947.52,
      "licensed": true,
      "datedded": "2018-09-18"
    }
  },
  {
    "id": 2,
    "location": {
      "name": "East wing",
      "warehouse": "Warehouse B",
      "lat": "15.95386",
      "long": "7.06246"
    },
    "carInfo": {
      "make": "Maserati",
      "model": "Coupe",
      "yearModel": 2005,
      "price": 19957.71,
      "licensed": false,
      "datedded": "2017-11-14"
    }
  },
  {
    "id": 3,
    "location": {
      "name": "Suid wing",
      "warehouse": "Warehouse C",
      "lat": "39.12788",
      "long": "-2.71398"
    },
    "carInfo": {
      "make": "Cadillac",
      "model": "Escalade",
      "yearModel": 2005,
      "price": 7429.18,
      "licensed": true,
      "datedded": "2018-09-26"
    }
  },
  {
    "id": 4,
    "location": {
      "name": "Suid wing",
      "warehouse": "Warehouse A",
      "lat": "-70.84354",
      "long": "132.22345"
    },
    "carInfo": {
      "make": "Saab",
      "model": "900",
      "yearModel": 1987,
      "price": 8771.0,
      "licensed": false,
      "datedded": "2017-12-01"
    }
  }
];
